components/net/tests/filemanager_thread.rs

```diff
@@ -2,31 +2,20 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use create_embedder_proxy;
 use ipc_channel::ipc;
-use net::filemanager_thread::{FileManager, UIProvider};
+use net::filemanager_thread::FileManager;
 use net_traits::blob_url_store::BlobURLStoreError;
 use net_traits::filemanager_thread::{FilterPattern, FileManagerThreadMsg, FileManagerThreadError, ReadFileProgress};
+use servo_config::prefs::{PrefValue, PREFS};
 use std::fs::File;
 use std::io::Read;
 use std::path::PathBuf;
 
-pub const TEST_PROVIDER: &'static TestProvider = &TestProvider;
-
-pub struct TestProvider;
-
-impl UIProvider for TestProvider {
-    fn open_file_dialog(&self, _path: &str, _patterns: Vec<FilterPattern>) -> Option<String> {
-        Some("tests/test.jpeg".to_string())
-    }
-
-    fn open_file_dialog_multi(&self, _path: &str, _patterns: Vec<FilterPattern>) -> Option<Vec<String>> {
-        Some(vec!["tests/test.jpeg".to_string()])
-    }
-}
-
 #[test]
 fn test_filemanager() {
-    let filemanager = FileManager::new();
+    let filemanager = FileManager::new(create_embedder_proxy());
+    PREFS.set("dom.testing.htmlinputelement.select_files.enabled", PrefValue::Boolean(true));
```

Comment:
I'm wondering why this was not necessary before?

Reply:
Previously there was mock UI provider in filemanager thread does inject test files instead (https://github.com/servo/servo/pull/20480/files#diff-5a65a0caa6da8aa6e23da0a4101c6a86L19) regardless of testing prefs. Now changed code path removes UI provider from thread and always asks to embedder message, so setting up test to mock fixture for file selection.